charts/spire/charts/spire-server/templates/tornjak-config.yaml

```diff
@@ -8,11 +8,29 @@ data:
   server.conf: |
     server {
       spire_socket_path = "unix:///tmp/spire-server/private/api.sock" # socket to communicate with SPIRE server
-
+      {{- if eq (include "spire-tornjak.connectionType" .) "http" }}
       http {
         enabled = true # if true, opens HTTP server
-        port = "10000" # if HTTP enabled, opens HTTP listen port at container port 10000
+        port = "10080" # if HTTP enabled, opens HTTP listen port at specified container port
+      }
+      {{- end }}
+      {{- if eq (include "spire-tornjak.connectionType" .) "tls" }}
+      tls {
+        enabled = true
+        port = "10443" # container port for TLS connection
+        cert = "/opt/spire/server/tls.crt" # TLS server cert
+        key  = "/opt/spire/server/tls.key"  # TLS server key
+      }
+      {{- end }}
+      {{- if eq (include "spire-tornjak.connectionType" .) "mtls" }}
+      mtls {
+        enabled = true
+        port = "10443" # container port for mTLS connection
+        cert = "/opt/spire/server/tls.crt" # mTLS server cert
+        key  = "/opt/spire/server/tls.key"  # mTLS server key
+        ca   = "/opt/spire/user/ca.crt"  # mTLS user CA
```

Comment:
I would recommend simplifying this to 

```
http {
  port = 10080
}
https {
  port     = 10443
  cert     = …
  key      = …
  clientCA = … (optional setting)
}
```

Implementation wise you will do a permanent redirect if https section is in the config from `http` to `https`.

https section requires mandatory the cert and key.

if https section also has the clientCA, then you will verify that a client Cert is provided.

This way both install and user time the entire connection type is unobtrusive to the end user.

Basically only thing they need to know is the endpoint and all the rest is transparantly resolved.